ADQL.tex

```diff
@@ -289,7 +289,7 @@ \subsubsection{ADQL reserved keywords}
 \texttt{EXP,} \texttt{FLOOR,} \texttt{LOG,} \texttt{LOG10,} 
 \texttt{MOD,} \texttt{PI,} \texttt{POWER,} \texttt{RADIANS,} 
 \texttt{RAND,} \texttt{ROUND,} \texttt{SIN,} \texttt{SQRT,} 
-\texttt{TAN,} \texttt{TOP,} \texttt{TRUNCATE}
```

Comment:
This doesn't seem to have ended up elsewhere? Is it still needed? Does its removal affect the discussion in 2.1.2 para 3?

Reply:
It is because I do not consider `TOP` really as a mathematical function or operator. It is not related to math. It just limits the number of rows in the query result. I do not know why it was here in the first place.

Anyway, `TOP` is in the BNF. As many other operators or clauses keywords that are neither in the list of SQL reserved keywords nor in this ADQL reserverd keywords section, or only in the SQL reserved keywords....it seems there is a mess here since the 2.0. That's why, I added a comment in the commit message stating that the reserved keywords in the BNF and in the text, should be reviewed but that would be done anway in the next version of ADQL when replacing the BNF (broken) grammar by a PEG one (machine readable, and so I hope, much more reliable). So, I don't know if it worth making both the BNF and the text really exhaustive on reserved keywords in this present version. I just tried to list the most important one ; the details being in the BNF and the entire ADQL grammar anyway.

But if anyone thinks that it should exhaustive and detailed in both parts, just tell me and I'll do it. It may be more clean though probably a bit more long-ish.

Reply:
> But if anyone thinks that it should exhaustive and detailed in both parts, just tell me and I'll do it. It may be more clean though probably a bit more long-ish.

Fix it in the PEG grammar in version 3.x.

Reply:
Pushing this to the PEG sounds good to me.